lib/task.js

```diff
@@ -599,10 +598,11 @@ export default class Task {
       return await this.abortRun(this.taskState);
     }
     // Validate the schema!
-    let payloadErrors =
-      this.runtime.schema.validate(this.task.payload, PAYLOAD_SCHEMA);
+    let payloadErrors = await validator().then(function (validator) {
+      return validator.check(this.task.payload, payloadSchema);
```

Comment:
Ah, it might be.  using arrow functions now makes this a lot less painful as it always captures the context of the surrounding scope.